src/Database/LSMTree/Internal.hs

```diff
@@ -1278,3 +1310,115 @@ duplicate t@Table{..} = do
             tableConfig
             tableArenaManager
             content
+
+
+{-------------------------------------------------------------------------------
+   Table union
+-------------------------------------------------------------------------------}
+
+{-# SPECIALISE union :: Table IO h -> Table IO h -> IO (Table IO h) #-}
+-- | See 'Database.LSMTree.Normal.union'.
+union ::
+     (MonadMask m, MonadMVar m, MonadST m, MonadSTM m)
+  => Table m h
+  -> Table m h
+  -> m (Table m h)
+union t1 t2 = unions $ t1 :| [t2]
+
+{-# SPECIALISE unions :: NonEmpty (Table IO h) -> IO (Table IO h) #-}
+-- | See 'Database.LSMTree.Normal.unions'.
+unions ::
+     (MonadMask m, MonadMVar m, MonadST m, MonadSTM m)
+  => NonEmpty (Table m h)
+  -> m (Table m h)
+unions ts = do
+    conf <-
+      case match (fmap tableConfig ts) of
+        Left (i, j) -> throwIO $ ErrUnionsTableConfigMismatch i j
+        Right conf  -> pure conf
```

Comment:
Do we really need the configs to match? It's okay as a requirement for now, but it should be documented in the public API and maybe have a TODO that we might want to revisit it.

Most settings don't really need to match for things to work, but of course we'd still need to answer the question of which config to use for the new table, possibly requiring to supply it manually?

Reply:
This could always be generalised later to have a config compatibility test and config merge.

Reply:
Valid point, I'll add a todo

Reply:
FYI the fact that the configs have to match is already specified in the public API haddocks